### PR TITLE
[Merged by Bors] - feat(UniformEmbedding): add `UniformInducing.completeSpace_congr`

### PR DESCRIPTION
**Pull request by urkud**
This is a version of `_root_.completeSpace_congr`
that applies to uniform inducing quotient maps.
Also add `ULift.completeSpace_iff` and `UniformInducing.completeSpace`.

I'm going to use these lemmas to generalize some `CompleteSpace` instances,
see #17244.

---

I also golfed some proofs that were broken by a change that I later reverted.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
Skimming the diff, it's not immediately obvious which lemma you generalized. Please name it in the PR description!

**Comment by urkud** [comment]
@eric-wieser I've just reverted generalization of `_root_.completeSpace_congr`. I'll update PR description in a few minutes.

File: Mathlib/Topology/UniformSpace/UniformEmbedding.lean [hunk 1]
```diff
@@ -286,38 +286,44 @@ theorem completeSpace_iff_isComplete_range {f : α → β} (hf : UniformInducing
     CompleteSpace α ↔ IsComplete (range f) := by
   rw [completeSpace_iff_isComplete_univ, ← isComplete_image_iff hf, image_univ]
 
+alias ⟨_, UniformInducing.completeSpace⟩ := completeSpace_iff_isComplete_range
+
 theorem UniformInducing.isComplete_range [CompleteSpace α] {f : α → β} (hf : UniformInducing f) :
     IsComplete (range f) :=
   (completeSpace_iff_isComplete_range hf).1 ‹_›
 
+theorem UniformInducing.completeSpace_congr {f : α → β} (hf : UniformInducing f)
```
Comment (eric-wieser) [inline Mathlib/Topology/UniformSpace/UniformEmbedding.lean:295]: Can you add a docstring comparing this to `_root_.completeSpace_congr`, and perhaps one in the other direction too?

File: Mathlib/Topology/UniformSpace/UniformEmbedding.lean [hunk 2]
```diff
@@ -286,38 +286,44 @@ theorem completeSpace_iff_isComplete_range {f : α → β} (hf : UniformInducing
...
+  rw [completeSpace_iff_isComplete_range uniformEmbedding_subtype_val.toUniformInducing,
+    Subtype.range_coe]
 
 alias ⟨_, IsComplete.completeSpace_coe⟩ := completeSpace_coe_iff_isComplete
 
 theorem IsClosed.completeSpace_coe [CompleteSpace α] {s : Set α} (hs : IsClosed s) :
     CompleteSpace s :=
   hs.isComplete.completeSpace_coe
 
+theorem ULift.completeSpace_iff : CompleteSpace (ULift α) ↔ CompleteSpace α :=
```
Comment (eric-wieser) [inline Mathlib/Topology/UniformSpace/UniformEmbedding.lean:321]: Or maybe `completeSpace_ulift_iff`, to match things like `completeSpace_coe_iff_isComplete`

**Review by eric-wieser — APPROVED** [review]
bors d+

File: Mathlib/Topology/UniformSpace/UniformEmbedding.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Topology/UniformSpace/UniformEmbedding.lean:321]: Opened a [Zulip poll](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Naming.3A.20.60Typeclass.20.28ULift.20.CE.B1.29.20.E2.86.94.20Typeclass.20.CE.B1.60)

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**